manta-pay/abstract-protocol.tex

```diff
@@ -328,100 +350,140 @@ \subsection{Abstract Cryptographic Schemes}
 \begin{definition}[\Transfer{} Object]
     A \Transfer{} is the following tuple:
     \begin{align*}
-        \sources    &: \List(\Asset) \\
-        \senders    &: \List(\Sender) \\
-        \receivers  &: \List(\Receiver) \\
-        \sinks      &: \List(\Asset)
+        \ID        &: \Option(\AssetId) \\
+        \sources   &: \List(\AssetValue) \\
+        \senders   &: \List(\Sender) \\
+        \receivers &: \List(\Receiver) \\
+        \sinks     &: \List(\AssetValue)
     \end{align*}
     The \emph{shape} of a \Transfer{} is the following $4$-tuple of cardinalities of those sets
     \[\left(\abs{T.\sources}, \abs{T.\senders}, \abs{T.\receivers}, \abs{T.\sinks}\right)\]
+    Also, note that the $\ID$ value is optional. This is inhabited whenever there are $\sources$ or $\sinks$, but if the shape of the transaction is $(0, m, n, 0)$ then $\ID = \None$.
 \end{definition}
 
 In order for a \Transfer{} to be considered \emph{valid}, it must adhere to the following constraints:
 
 \begin{itemize}
+    \item \textbf{Correct Key Signing}: The keys used to construct \Sender{s} and \Receiver{s} are valid and can be signed by a unique \SpendingKey{}.
```

Comment:
Is the `SpendingKey` here the same as the ephemeral secret key `esk`?

Reply:
The spending key is the authority for signing the transaction. The ephemeral secret key is just for encryption.